src/Database/DBA.php

```diff
@@ -669,6 +669,10 @@ public static function collapseCondition(array $condition)
 	 */
 	public static function mergeConditions(array ...$conditions)
 	{
+		if (count($conditions) == 1) {
+			return current($conditions);
+		}
```

Comment:
This isn't the condition you want in this case. Instead, you should add the following condition in the following `foreach`:
```php
if (!$condition) {
    continue;
}
```